---
title: "Transcriptomics and the analysis of RNA-Seq data"
author: "Sunyoung Lee"
date: "February 21, 2020"
output: github_document
---

##Read out data for today

```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("airway_metadata.csv", stringsAsFactors = FALSE)
```

```{r}
head(counts)
```
How many genes are we dealing with?


```{r}
nrow(counts)
```

We will compare the controls vs treated count data contained in `counts` object. We will first extract the control columns by looking up their colnames in the `metadata` object. 

```{r}
x<-1:5
x>3
x[x>3]
```




```{r}
#metadata
#metadata$dex == "control"
#Access just the "control" metadata 
control <- metadata[metadata$dex == "control",]
control
```

```{r}
#Access the counts columns with control$id 
counts[,control$id]
```

```{r}
#Now find the mean value for each gene (i.e. each row)
control.mean <- rowSums(counts[,control$id]) / nrow(control)
names(control.mean) <- counts$ensgene
control.mean
```

```{r}
treated <- metadata[metadata$dex == "treated",]
treated
```

```{r}
counts[,treated$id]

```

```{r}
treated.mean <- rowSums(counts[,treated$id]) / nrow(treated)
names(treated.mean) <- counts$ensgene
treated.mean
```

Let's put these together for ease of use

```{r}
mycounts <- data.frame(control.mean, treated.mean)
```


```{r}
plot(mycounts)
```

```{r}
hist(mycounts$control.mean, breaks=300)
```

Our data is very skewed so lets with with logs. First re-plot our control vs treated on a log-log

```{r}
plot(mycounts, log="xy")
```

Let's remove our zero count genes because we can't say aything about them from this dataset!

```{r}
x <- data.frame(c(1,3,10,0), c(1,3,0,0))
x
unique(which(x==0, arr.ind=TRUE)[,"row"])
```

Ok find zero rows in our counts object

```{r}
to.rm <- unique(which(mycounts == 0, arr.ind=TRUE)[,"row"])
newcounts <- mycounts[-to.rm, ]
newcounts
```

How many genes have we left?

```{r}
nrow(newcounts)
```

```{r}
plot(newcounts, log="xy")

```
```{r}
newcounts$log2fc <- log2(newcounts[,"treated.mean"]/newcounts[,"control.mean"])
head(newcounts)
```

A common rule of thumb in the filed is to use a log2fc of greater than +2 as *upregulated* and less than -2 as *downregulated*

```{r}
upregulated <- sum(newcounts$log2fc > 2)
upregulated
```

```{r}
downregulated <- sum(newcounts$log2fc < -2)
downregulated
```

Let's use DESeq2

```{r}
library(DESeq2)

```

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

Volcano plots

Plot of log2fc vs p-value
```{r}
plot(res$log2FoldChange, -log(res$padj), col="gray")
abline(v=c(-2,+2), lty=2)
abline(h=-log(0.05), lty=2)
```

